kornia/augmentation/_2d/intensity/plasma.py

```diff
@@ -0,0 +1,150 @@
+from typing import Dict, Optional, Tuple
+
+from torch import Size, Tensor
+from torch.distributions import Uniform
+
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.contrib import diamond_square
+
+
+class RandomPlasmaBrightness(IntensityAugmentationBase2D):
+    r"""Adds brightness to the image based on a fractal map generated by the diamond square algorithm.
+
+    .. image:: _static/img/RandomPlasmaBrightness.png
+
+    This is based on the original paper: TorMentor: Deterministic dynamic-path, data augmentations with fractals.
+    See: :cite:`tormentor` for more details.
+
+    .. note::
+        This function internally uses :func:`kornia.contrib.diamond_square`.
+
+    Args:
+        roughness: value to scale during the recursion in the generation of the fractal map.
+        intensity: value that scales the intensity values of the generated maps.
+        p: probability of applying the transformation.
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> img = torch.ones(1, 1, 3, 4)
+        >>> RandomPlasmaBrightness(roughness=(0.1, 0.7), p=1.)(img)
+        tensor([[[[0.6415, 1.0000, 0.3142, 0.6836],
+                  [1.0000, 0.5593, 0.5556, 0.4566],
+                  [0.5809, 1.0000, 0.7005, 1.0000]]]])
+    """
+
+    def __init__(
+        self, roughness: Tuple[float, float] = (0.1, 0.7), intensity: Tuple[float, float] = (0.0, 1.0), p: float = 0.5
+    ) -> None:
+        super().__init__(p=p)
+        self.roughness = Uniform(roughness[0], roughness[1])
+        self.intensity = Uniform(intensity[0], intensity[1])
+
+    def generate_parameters(self, shape: Size) -> Dict[str, Tensor]:
+        B = shape[0]  # batch_size
+        roughness = self.roughness.rsample((B,))
+        intensity = self.intensity.rsample((B, 1, 1, 1))
+        return dict(roughness=roughness, intensity=intensity)
+
+    def apply_transform(self, image: Tensor, params: Dict[str, Tensor], transform: Optional[Tensor] = None) -> Tensor:
```

Comment:
@shijianjian i have an api suggestion here, it’s quite annoying to have to include the transform as empty for the intensity augmentations. Can we somehow remove it at least for intensity ?

Reply:
Not clear yet. Removing this is easy, but then more `if-else` will be needed for those sequentials, etc. Probably after supporting the non-affine transformations. Those APIs can be considered in more generic vision.

Reply:
Awesome